Rename to PyLCM (#113)
OpenSourceEconomics/pylcm: README.md

# Life Cycle Models

[![Ruff](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/astral-sh/ruff/main/assets/badge/v2.json)](https://github.com/astral-sh/ruff)
[![pre-commit.ci status](https://results.pre-commit.ci/badge/github/OpenSourceEconomics/pylcm/main.svg)](https://results.pre-commit.ci/latest/github/OpenSourceEconomics/pylcm/main)
[![image](https://codecov.io/gh/OpenSourceEconomics/pylcm/branch/main/graph/badge.svg)](https://codecov.io/gh/OpenSourceEconomics/pylcm)

This package aims to generalize and facilitate the specification, solution, and
simulation of finite-horizon discrete-continuous dynamic choice models.

> ## Installation

Author: I'd like to change the package dependencies in a separate PR. I would update the installation instructions there.

> ## Installation

Reviewer: Cool. Would just like to have the higher version number in here.

# Life Cycle Models

[![Ruff](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/astral-sh/ruff/main/assets/badge/v2.json)](https://github.com/astral-sh/ruff)

> [![pre-commit.ci status](https://results.pre-commit.ci/badge/github/OpenSourceEconomics/pylcm/main.svg)](https://results.pre-commit.ci/latest/github/OpenSourceEconomics/pylcm/main)

Reviewer: ```suggestion
[![pre-commit.ci status](https://results.pre-commit.ci/badge/github/OpenSourceEconomics/PyLCM/main.svg)](https://results.pre-commit.ci/latest/github/OpenSourceEconomics/PyLCM/main)
```
Doesn't matter, ofc, but if we CamelCase OpenSourceEconomics, looks better if we do the same for PyLCM. Same below.

Author: Since we had mixed-use cases of lowercase and CamelCase for OpenSourceEconomics itself, I have decided to just do everything in lowercase. Seems to be more in line with, e.g., PyTorch.

[![image](https://codecov.io/gh/OpenSourceEconomics/pylcm/branch/main/graph/badge.svg)](https://codecov.io/gh/OpenSourceEconomics/pylcm)

This package aims to generalize and facilitate the specification, solution, and
simulation of finite-horizon discrete-continuous dynamic choice models.

## Installation

`PyLCM` currently needs to be installed from GitHub and requires `jax` and `jaxlib`.

> [!IMPORTANT]

> > We recommend installing at least version "0.4.34" of `jax` and `jaxlib`, due to memory

Reviewer: ```suggestion
> We recommend installing at least version "0.5.2" of `jax` and `jaxlib`, due to memory
```

> issues with previous versions.

If you aim to run `PyLCM` on a GPU, you need to install `jaxlib` with CUDA support (for
Linux) or with support for AMD GPUs / ARM-based Silicon GPUs (for MacOS). In any case,
for installation of `jax` and `jaxlib`, please consult the `jax`
[docs](https://jax.readthedocs.io/en/latest/installation.html#supported-platforms).

> [!NOTE]
> We currently test GPU support for `PyLCM` only on Linux with CUDA 12.

## Questions

If you have any questions, feel free to ask them on the PyLCM
[Zulip chat](https://ose.zulipchat.com/#narrow/channel/491562-PyLCM).

## License

This project is licensed under the Apache License, Version 2.0. See the
[LICENSE](LICENSE) file for details.

> Copyright (c) 2023 The PyLCM Authors

Reviewer: ```suggestion
Copyright (c) 2023- The PyLCM Authors
```
